### DISCUSSION TITLE
React Query Caching not updating after cache time

### DISCUSSION THREAD

**Discussion by Karthick1911** [TanStack/query#1256] [Q&A]
Hi,

I'm new to react query, i have set the below as default config
`defaultConfig: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },`
and set the cacheTime to 10millisec. Even after 10 millisec if i update the component I get the same cached data even if new data is added to api by other members. Wats the actuall usage of using cacheTime.

Crct me if my understanding is wrong.

**Comment by TkDodo** [comment]
React-query doesn’t re-fetch when you render a component. It’s well described here: https://react-query.tanstack.com/docs/guides/important-defaults

cacheTime only sets the time until inactive queries (those that have no observer) are being removed from the cache. You want staleTime, but that is 0 anyways. 

Not sure what your use-case is exactly. Maybe you want to set a `refetchInterval` to fetch periodically, i don’t know..

**Comment by Karthick1911** [comment]
Can I know wat happens when queries are inactive, I tried setting cacheTime
to 10ms but after 10ms to I can able to get the data from cache is that the
default behavior or am I wrong

**Reply by TkDodo** [reply]
cacheTime means: when this query is inactive (which means it has no observers, i.e. all components that use this query are unmounted), it will stay in the cache for this time before it will be garbage collected.
